app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt

```diff
@@ -1,46 +1,316 @@
+
+              // Display the workout sessions
+              WorkoutSessionsSection(
+                  workouts = workouts, profile = profile, navigationActions = navigationActions)
+
+              // Display the quick workout section
+              QuickWorkoutSection(navigationActions = navigationActions)
+
+              // Display the new workout plan button
+              NewWorkoutSection(navigationActions = navigationActions)
             }
       },
-      bottomBar = {
-        BottomNavigationMenu(
-            onTabSelect = { route -> navigationActions.navigateTo(route) },
-            tabList = LIST_OF_TOP_LEVEL_DESTINATIONS,
-            selectedItem = navigationActions.currentRoute())
-      })
+      bottomBar = { BottomNavigationBar(navigationActions = navigationActions) })
+}
+
+// Composable function for the profile section with profile picture and settings
+@Composable
+fun ProfileSection(account: UserAccount, profile: Int, navigationActions: NavigationActions) {
+  Row(
+      modifier =
+          Modifier.fillMaxWidth()
+              .background(DarkBlue)
+              .padding(vertical = 16.dp)
+              .testTag("ProfileSection")) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier.padding(horizontal = 16.dp)) {
+              // Profile image
+              Image(
+                  painter = painterResource(id = profile),
+                  contentDescription = "Profile",
+                  modifier = Modifier.size(48.dp).clip(CircleShape).testTag("ProfilePicture"))
+              Spacer(modifier = Modifier.width(8.dp))
+              // Display user name dynamically
+              Text(
+                  text = "What's up ${account.firstName}?",
+                  style = MaterialTheme.typography.titleSmall.copy(fontSize = 30.sp),
+                  color = Color.White,
+                  modifier = Modifier.testTag("WelcomeText"))
+            }
+
+        Spacer(Modifier.weight(1f))
+
+        // Settings Icon
+        IconButton(
+            onClick = { navigationActions.navigateTo(Screen.SETTINGS) },
+            modifier = Modifier.padding(end = 16.dp).testTag("SettingsButton")) {
+              Icon(
+                  imageVector = Icons.Outlined.Settings,
+                  contentDescription = "Settings",
+                  tint = Color.White,
+                  modifier = Modifier.size(30.dp))
+            }
+      }
+}
+
+// Composable function for workout sessions section
+@Composable
+fun WorkoutSessionsSection(
+    workouts: List<Workout>,
+    profile: Int,
+    navigationActions: NavigationActions
+) {
+  Column(modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp).testTag("WorkoutSection")) {
+    Text(
+        text = "My workout sessions",
+        style = MaterialTheme.typography.titleSmall.copy(fontSize = 25.sp),
+        modifier = Modifier.padding(vertical = 8.dp))
+
+    Column(
```

Comment:
I would suggest using a Lazy Column instead of Column, as in the future we might want to be able to have more workouts. This would make the UI rendering a bit more efficient.